From: Alexandra Winter <wintera@linux.ibm.com>
Date: Thu, 23 Oct 2025 17:06:35 +0200
Subject: [PATCH] dibs: Remove reset of static vars in dibs_init()
References: bsc#1258210
Patch-mainline: 6.18.53
Git-commit: 182663bbff784e88cf7acbae7c53c6c5e0cb9ca4

[ Upstream commit 182663bbff784e88cf7acbae7c53c6c5e0cb9ca4 ]

'clients' and 'max_client' are static variables and therefore don't need to
be initialized.

Reported-by: Mete Durlu <meted@linux.ibm.com>
Signed-off-by: Alexandra Winter <wintera@linux.ibm.com>
Reviewed-by: Alexander Lobakin <aleksander.lobakin@intel.com>
Reviewed-by: Dust Li <dust.li@linux.alibaba.com>
Link: https://patch.msgid.link/20251023150636.3995476-1-wintera@linux.ibm.com
Signed-off-by: Paolo Abeni <pabeni@redhat.com>
Stable-dep-of: 1668a31e3b1a ("dibs: Unregister dibs_class after error")
Signed-off-by: Sasha Levin <sashal@kernel.org>
Signed-off-by: Robert Frohl <rfrohl@suse.com>
---
 drivers/dibs/dibs_main.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/dibs/dibs_main.c b/drivers/dibs/dibs_main.c
index cf96f0a4a5af..d6ea178e70c4 100644
--- a/drivers/dibs/dibs_main.c
+++ b/drivers/dibs/dibs_main.c
@@ -248,9 +248,6 @@ static int __init dibs_init(void)
 {
 	int rc;
 
-	memset(clients, 0, sizeof(clients));
-	max_client = 0;
-
 	dibs_class = class_create("dibs");
 	if (IS_ERR(dibs_class))
 		return PTR_ERR(dibs_class);
-- 
2.53.0

